common/cache.py

```
@@ -0,0 +1,190 @@
+
+        Returns:
+            Response or None: The cached response if found, otherwise None.
+        """
+        cached_data = cache.get(cache_key)
+        if cached_data is not None:
+            logger.debug(
+                f"Cache Hit for {self.primary_model.__name__} - Cache Key: {cache_key}"
+            )
+            metrics.increment_cache(model=self.primary_model.__name__, type="hit")
+            return Response(cached_data, status=status.HTTP_200_OK)
+        else:
```

Ruff (0.12.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)